### DISCUSSION TITLE
Routing question with dynamic existing URLs

### DISCUSSION THREAD

**Discussion by ndebarnot** [symfony/symfony#44328] [Q&A] [answered]
Hello,

I'm considering moving a big existing application to Symfony but I'm not sure the routing might work as I expect and I would like to keep my existing URLs as much as possible. This can be a no-go for me so I'm turning to experts before going any further.

Our application is a CMS. We have different types of content, for instance news and articles and it's all organized in categories, brands and products.

Existing URLs are as follows:

- /first-category/news/ => lists all news in "first-category"
- /second-category/news/ => lists all news in "second-category"
- /brand-1/news/ => lists all news about "brand-1"
- /brand-2/news/ => lists all news about "brand-2"
but it can also be something like this:
- /first-category/brand-2/news/ => lists all news about "brand-2" in "first-category"
or even like this:
- /second-category/brand-1/product-4/ => lists all news about "product-4" which belongs to "brand-1" and is in "second-category"

We can have exactly the same thing for the "article" type of content.

All such URLs with /news/ should go to the NewsController::list() whereas all such URLs with /articles/ should go to ArticleController::list().

Is something like this possible with Symfony?

Thanks a lot for your help!

**Comment by stof** [comment, marked as answer]
> All such URLs with /news/ should go to the NewsController::list() whereas all such URLs with /articles/ should go to ArticleController::list().
>
> Is something like this possible with Symfony?

Yes it is.
You might need to define multiple routes for news to reproduce the URL scheme (the case of having both a brand and a category probably needs to use a different route than when having only one) but it is totally valid to have multiple routes pointing to the same controller.

To me, the more complex will probably be to distinguish `/first-category/news/` and `/brand-1/news/` to know whether the first segment is a brand or a category. If this requires checking the DB for that, you might need to have a single route passing an unknown first segment that is then identified by the controller.

**Reply by ndebarnot** [reply]
Thanks a lot! Did not know that it was possible to have multiple routes for an action.

Distinguishing categories and brands should not be a problem, this we already do and yes, it requires a database query.

Thank you, I'll go forward with my investigation then ;)
